apps/desktop-wallet/src/api/apiDataHooks/address/useFetchAddressTokensByType.ts

```diff
@@ -1,18 +1,23 @@
+import { useCurrentlyOnlineNetworkId } from '@alephium/shared-react'
+import { useQuery } from '@tanstack/react-query'
+
 import { UseFetchAddressProps } from '@/api/apiDataHooks/address/addressApiDataHooksTypes'
-import useFetchAddressBalances from '@/api/apiDataHooks/address/useFetchAddressBalances'
-import useFetchTokensSeparatedByType from '@/api/apiDataHooks/utils/useFetchTokensSeparatedByType'
+import { addressTokensByTypeQuery } from '@/api/queries/addressQueries'
 
-interface UseFetchAddressTokensByType extends UseFetchAddressProps {
-  includeAlph: boolean
-}
+const useFetchAddressTokensByType = ({ addressHash }: UseFetchAddressProps) => {
+  const networkId = useCurrentlyOnlineNetworkId()
 
-const useFetchAddressTokensByType = (props: UseFetchAddressTokensByType) => {
-  const { data: allTokensBalances, isLoading: isLoadingBalances } = useFetchAddressBalances(props)
-  const { data, isLoading } = useFetchTokensSeparatedByType(allTokensBalances)
+  const { data, isLoading } = useQuery(addressTokensByTypeQuery({ addressHash, networkId }))
```

Comment:
Utilizing Tanstack query cache to avoid re-calculating data by moving code into a `queryFn`, like we currently do in other areas.